src/olympia/reviewers/decorators.py

```diff
@@ -35,18 +37,19 @@ def wrapper(request, *args, **kw):
 
 
 def unlisted_addons_reviewer_required(f):
-    """Require an "unlisted addons" reviewer user.
+    """Require an "unlisted addons" reviewer or viewer user.
 
-    The user logged in must be an unlisted addons reviewer or admin.
-
-    An unlisted addons reviewer is someone who is in a group with the following
-    permission: 'Addons:ReviewUnlisted'.
+    The user logged in must satisfy one of the following:
+    - be an admin
+    - have the permission 'ReviewerTools:ViewUnlisted' (unlisted viewer)
+    - have the permission 'Addons:ReviewUnlisted' (unlisted reviewer)
     """
 
     @login_required
     @functools.wraps(f)
     def wrapper(request, *args, **kw):
-        if acl.check_unlisted_addons_reviewer(request):
+        view_on_get = _view_on_get(request)
+        if view_on_get or acl.check_unlisted_addons_reviewer(request):
```

Comment:
This would allow someone with just `REVIEWER_TOOLS_VIEW` to view a page decorated by `unlisted_addons_reviewer_required` since you modified `_view_on_get` to allow this. I think you shouldn't change `_view_on_get` and do something specific here for `REVIEWER_TOOLS_UNLISTED_VIEW`.

Reply:
Fixed in 200de0dddb403472977c3edb75bed9e91730f791.